packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -682,7 +697,15 @@ class EditableText extends StatefulWidget {
        assert(rendererIgnoresPointer != null),
        assert(scrollPadding != null),
        assert(dragStartBehavior != null),
+       assert(
+          (contentInsertionMimeTypes != null && onContentInserted != null) ||
+          contentInsertionMimeTypes == null ||
+          onContentInserted == null
+       ),
        enableInteractiveSelection = enableInteractiveSelection ?? (!readOnly || !obscureText),
+       contentInsertionMimeTypes = contentInsertionMimeTypes ??
```

Comment:
`nil` and empty `List` do not mean the same thing:

`nil`: Tells engine that `contentInsertionMimeTypes` was not provided, so it should fall back to the default mime types (ie `kDefaultMimeTypes`

empty `List`: Tells engine to fully disable content insertion by setting no mime types. This is useful because then the OS will show a toast saying that the text field doesn't support content insertion, so the user knows they cannot use image keyboard features there.

Reply:
That might be something useful to explain in the docs for onContentInserted and contentInsertionMimeTypes.

Reply:
Then maybe make it non-nullable (`List<String> contentInsertionMimeTypes = kDefaultMimeType`) instead?

Reply:
I made it non-nullable with default empty value. If you see the [docs](https://github.com/flutter/flutter/pull/110052/files#diff-f5f93c879cef9a102adbf148583de9f3d9a05f9678fdcad7bde106f62200bb34R1622-R1628) for contentInsertionMimeTypes the approach should make more sense.